Tasks/google-play-promote/googleutil.ts

```diff
@@ -0,0 +1,227 @@
+// common code shared by all tasks
```

Comment:
This file is duplicated 3x.  There may be a way to share the code across tasks if I demand a higher version of Node.  I may look into that as a future change.

Reply:
I don't understand why you can share the code due to the node version, can you summarize? Can you pull just the interface types out into a `.d.ts` file in a common location?

Reply:
The pattern we use for sharing common code is to create a 'common package'.  See [this](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tasks/Common) for some examples.  When I refactored to use with that pattern, one of the modules I included required that I upgrade the version of TypeScript that's used (why? I don't really know), which cascaded into a couple of errors that showed up only during runtime, which I eventually traced to [this](https://github.com/googleapis/gaxios/issues/57) issue.  (That's over-simplifying things as there were a half-dozen other errors to work through to get to that one.) For whatever reason, using `googleapis` with a common library requires bumping the version of TypeScript that gets used, which requires many of the dependencies to bump their versions, which leads to a series of runtime errors.

To answer your question, there's likely a way to use another pattern and share code, but exactly what that way should be is not really clear to me.  After much time spent trying to get it to work the 'right' way, I downshifted to code duplication.  It was already duplicated across all the tasks anyway, so this isn't 'worse' than before.

The existing build scripts don't seem to me to allow a common build folder to be included, but if you have a suggestion on how to do it, I'd be happy to hear it.

Reply:
Gotcha (mostly).  I'll take your word for it, and I agree, no worse than where it was at before.

Reply:
'mostly' ... pretty much me too.  :)

Reply:
I'd argue that this is better than a "common" folder. We've had a lot of issues with that in the other tasks repo as it makes it harder to update individual tasks that all rely on the same common files (since a common folder has no versioning)